fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/TextIndexMaintainer.java

```diff
@@ -523,14 +524,21 @@ public boolean canDeleteWhere(@Nonnull QueryToKeyMatcher matcher, @Nonnull Key.E
         TextSubspaceSplitter subspaceSplitter = new TextSubspaceSplitter(state.indexSubspace, textPosition + 1);
         Range byteRange = range.toRange();
         ScanProperties withAdjustedLimit = scanProperties.with(ExecuteProperties::clearSkipAndAdjustLimit);
+        ExecuteProperties adjustedExecuteProperties = withAdjustedLimit.getExecuteProperties();
         BunchedMapMultiIterator<Tuple, List<Integer>, Tuple> iterator = BUNCHED_MAP.scanMulti(
-                state.context.readTransaction(scanProperties.getExecuteProperties().getIsolationLevel().isSnapshot()),
+                state.context.readTransaction(adjustedExecuteProperties.getIsolationLevel().isSnapshot()),
                 state.indexSubspace,
                 subspaceSplitter,
                 byteRange.begin,
                 byteRange.end,
                 continuation,
-                withAdjustedLimit.getExecuteProperties().getReturnedRowLimit(),
+                adjustedExecuteProperties.getReturnedRowLimit(),
+                keyValue -> {
+                    final ByteScanLimiter byteScanLimiter = adjustedExecuteProperties.getState().getByteScanLimiter();
+                    if (byteScanLimiter != null) {
+                        byteScanLimiter.registerScannedBytes(keyValue.getKey().length + keyValue.getValue().length);
+                    }
+                },
```

Comment:
I suppose this could check if the `byteScanLimiter` is non-null prior to constructing the iterator and then set the callback to `null` (which I believe is supported by the API) if it is or call back that always calls `registerScannedBytes` if it isn't. That (I think) saves an allocation if the bytes scan limiter isn't turned off, but it's probably not too big a deal.